client/workflow_job.go

```diff
@@ -0,0 +1,89 @@
+package awx
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+)
+
+// Enum of job statuses.
+const (
+	WorkflowJobStatusNew        = "new"
+	WorkflowJobStatusPending    = "pending"
+	WorkflowJobStatusWaiting    = "waiting"
+	WorkflowJobStatusRunning    = "running"
+	WorkflowJobStatusSuccessful = "successful"
+	WorkflowJobStatusFailed     = "failed"
+	WorkflowJobStatusError      = "error"
+	WorkflowJobStatusCanceled   = "canceled"
+)
+
+// WorkflowJobService implements awx workflow job apis.
+type WorkflowJobService struct {
+	client *Client
+}
+
+// CancelWorkflowJobResponse represents `CancelWorkflowJob` endpoint response.
+type CancelWorkflowJobResponse struct {
+	Detail string `json:"detail"`
+}
+
+const WorkflowJobAPIEndpoint = "/api/v2/workflow_jobs/"
+
+// GetWorkflowJob shows the details of a workflow job.
+func (j *WorkflowJobService) GetWorkflowJob(id int, params map[string]string) (*Job, error) {
```

Comment:
You use Job struct for workflowJob answer, but the two models are not the same I think.
https://docs.ansible.com/automation-controller/4.4/html/controllerapi/api_ref.html#/Workflow_Jobs/Workflow_Jobs_workflow_jobs_read

Reply:
@denouche Thank you for your feedback and suggestion. I have integrated new structs according to the documentation, specifically WorkflowJobLaunch and WorkflowJob, which accurately represent the API models as per the official documentation.